waku/factory/waku.nim

```diff
@@ -146,7 +148,32 @@ proc newCircuitRelay(isRelayClient: bool): Relay =
     return RelayClient.new()
   return Relay.new()
 
-proc new*(T: type Waku, confCopy: var WakuNodeConf): Result[Waku, string] =
+proc setupCallbacks(
+    node: WakuNode, conf: WakuNodeConf, callbacks: WakuCallbacks
+): Result[void, string] =
+  if callbacks.isNil():
+    info "No external callbacks to be set"
+    return ok()
+
+  if not callbacks.relayHandler.isNil():
```

Comment:
Super nitpick comment but I think the code gets a bit linear if:

```suggestion
  if callbacks.relayHandler.isNil():
     return err("error in setupCallbacks callbacks.relayHandler.isNil")
```

Reply:
The issue is that it isn't an error, and we also shouldn't return because the idea is to configure all the application level callbacks provided from that procedure. So if no application level callback is passed for Relay, then we don't configure it and will check for other ones (for example, the callback to run on topic health change).

It's a logic similar to when we try to setup and mount protocols

Reply:
What if we just use the following at the beginning?
```code
  if callbacks.isNil() or callbacks.relayHandler.isNil():
    info "No external callbacks to be set"
    return ok()
```

Reply:
but this doesn't scale well when more callbacks are provided, it would look like

```
if callbacks.isNil() or (callbacks.relayHandler.isNil() and callbacks.otherHandler1.isNil() and callbacks.otherHandler2.isNil(), ...):
    info "No external callbacks to be set"
    return ok()
```